### ISSUE TITLE
[HOLD for payment 2023-08-14] [$1000] Add Tooltip for the report actions whose actor is the workspace

### ISSUE THREAD

**Issue by cristipaval** [Expensify/App#21322]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to a workspace where you are a non-admin member
2. Request money from the workspace
3. Verify that you have the report preview where the actor is the workspace
4. Hover the workspace name on that report preview

## Expected Result:
You should see a tooltip with the workspace avatar and name

## Actual Result:
Nothing happens

## Workaround:
No workaround

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012a1fc3c444f423b4</li>
        <li>Upwork Job ID: 1675995747731927040</li>
        <li>Last Price Increase: 2023-07-10</li>
    </ul>
</details>

**Comment by cristipaval** [comment]
Hey @jliexpensify, this is a follow-up issue after [this PR](https://github.com/Expensify/App/pull/20941#issuecomment-1602865492). Don't try to reproduce this before the PR gets merged and deployed.

**Comment by abdulrahuman5196** [comment]
If required i can work on this issue as C+ in future since the original PR and this followup issue was found during the C+ review.

**Comment by cristipaval** [comment]
Yes, makes sense!

**Comment by jliexpensify** [comment]
Looks like the PR got deployed to Staging, so testing now.

**Comment by jliexpensify** [comment]
@cristipaval I'm confused by the instructions:

> 2. Request money from the workspace

How do you do this? Don't you request money from individuals?

> 3. Verify that you have the report preview where the actor is the workspace

What does "where the actor is the workspace". mean?

Thanks!

**Comment by cristipaval** [comment]
Hey @jliexpensify! The App PR which fixes this issue is already merged and deployed to staging. So you won't be able to reproduce the issue.

In order to understand the testing steps, it might help you to watch the screen recordings in the PR that fixes the issue, [here](https://github.com/Expensify/App/pull/20941#issue-1761299549).

**Comment by jliexpensify** [comment]
Had some additional questions, so reached out to Cristi via DM.

**Comment by cristipaval** [comment]
https://github.com/Expensify/App/assets/18078393/def7c1d7-e55c-4405-8885-39deb3be96d5

Hey @jliexpensify! This is how you request money from a workspace.

You can see that after requesting the money, you see in the chat the action "Workspace owes x money"

**Comment by jliexpensify** [comment]
Ok, turns out this was under a beta. Got myself added and now can test:

https://github.com/Expensify/App/assets/30815269/c5990db3-f3b4-49c2-8c22-496927a02613

**Comment by jliexpensify** [comment]
So I see the avatar and name, but this avatar is not correct:

![image](https://github.com/Expensify/App/assets/30815269/3beb4f86-b9fe-4318-81f8-9a7071643ea8)

cc @abdulrahuman5196

**Comment by abdulrahuman5196** [comment]
@jliexpensify No that was not the issue this GH is made for. I have marked in the below image and add video.


https://github.com/Expensify/App/assets/46707890/5fe0e97e-1f84-4579-aaee-708832defe51

<img width="1500" alt="Screenshot 2023-07-03 at 8 07 26 PM" src="https://github.com/Expensify/App/assets/46707890/039c7f2e-c7c7-4ede-b8ee-4c2848794ce5">

**Comment by jliexpensify** [comment]
Ok yeah, I can also see that - it would be nice to have a tooltip as we show one for individuals:

![image](https://github.com/Expensify/App/assets/30815269/78a282c8-9cf3-4826-94e7-f8f0911b894d)

But is this also an [issue](https://github.com/Expensify/App/issues/21322#issuecomment-1617374591)? For consistency's sake, we reflect the exact same display image for other tooltips:

![image](https://github.com/Expensify/App/assets/30815269/43a66abc-66fe-48bd-bf54-5d3e7aaaef16)

**Comment by jliexpensify** [comment]
@anmurali I am OOO until the end of the week, so just reassigning to help push this along. At the moment, we are waiting for proposals.

I'll unassign you when I'm back, thanks!

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
 Add Tooltip for the report actions whose actor is the workspace 
### What is the root cause of that problem?
This is feature request rather than a bug
### What changes do you think we should make in order to solve the problem?
We should update old tooltip to use `UserDetailsTooltip`, but we should modify it so that it can work with workspace data too
https://github.com/Expensify/App/blob/21bf80dedd544577d6c022176ef7ca7876ad0eff/src/components/SubscriptAvatar.js#L58

 We should introduce new prop to `UserDetailsTooltip` called `workspaceID`. Also we can create a helper function like this, which helps us get workspace avatar and name:
```javascript
function getPolicyAvatarAndName(policyID){
    const policy = lodashGet(allPolicies, `${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`);
    const name = lodashGet(policy, 'name')
    const avatar = lodashGet(policy, 'avatar') || getDefaultWorkspaceAvatar(name);
    
    return {name, avatar};
}
``` 
Based on this we can update data inside `UserDetailsTooltip`. Also, this is important, we should pass required fields into avatar inside tooltip since workspace's default avatar requires,  special props, so it can use correct fill color for letter avatars:
```javascript
                        type={props.workspaceID ? CONST.ICON_TYPE_WORKSPACE : CONST.ICON_TYPE_AVATAR}
                        name={userDisplayName}

```

We also need to implement this tooltip here
https://github.com/Expensify/App/blob/4382c2344463a0d96476d194c288641d664167b8/src/pages/home/report/ReportActionItemFragment.js#L152-L155
we can pass workspace id from `ReportActionItemSingle` like this:
```javascript
                                    workspaceID={isWorkspaceActor ? props.report.policyID : undefined}
``` 

### What alternative solutions did you explore? (Optional)

**Comment by anmurali** [comment]
@abdulrahuman5196 - this is pending your review

**Comment by anmurali** [comment]
Also pinged him on Slack - https://expensify.slack.com/archives/C049HHMV9SM/p1688750178448969

Going to unassign myself as @jliexpensify is back next week

**Comment by abdulrahuman5196** [comment]
@alitoshmatov / @Nodebrute Why do we want to introduce new different parameters? When UserDetailToolTip already has a fallback props exposed - https://github.com/Expensify/App/blob/main/src/components/UserDetailsTooltip/userDetailsTooltipPropTypes.js#L10
Shouldn't we just propogate it to parent? Or any other reason?

And @Nodebrute Kindly add code references from existing code to mention where you are expecting to make the change and how in your proposal.

**Comment by alitoshmatov** [comment]
@abdulrahuman5196 You are right, we can use `userFallbackData`, but we still need to inform tooltip that it is for workspace  to avatar work as expected when showing workspace avatar

**Comment by jliexpensify** [comment]
Bump @abdulrahuman5196 for reviews!

**Comment by jliexpensify** [comment]
Bump @abdulrahuman5196 - please review, thank you!

**Comment by abdulrahuman5196** [comment]
Reviewing the proposals now

**Comment by abdulrahuman5196** [comment]
Regarding @alitoshmatov 's proposal here https://github.com/Expensify/App/issues/21322#issuecomment-1620176373. I am not aligning with the proposal to create new param specific to workspaceID. I would like to keep the inner components as light weighted as possible. And since we also have other mechanisms to pass the data without this as well.

**Comment by abdulrahuman5196** [comment]
@Nodebrute 's proposal here https://github.com/Expensify/App/issues/21322#issuecomment-1619312058 looks good and works well. One other way i suggest to do is we can send prop called `actorIcon` to `ReportActionItemFragment` [here](https://github.com/Expensify/App/blob/main/src/pages/home/report/ReportActionItemFragment.js#L152) since we already have the icon [here](https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/pages/home/report/ReportActionItemSingle.js#L129) itself.

Note: Since basically I suggested the solution, I can't provide points on that to both the contributors. And as this is a straightforward implementation, I am choosing the first proposal with the most/similar proposal information.

🎀 👀 🎀
C+ Reviewed
cc: @cristipaval

**Comment by abdulrahuman5196** [comment]
Sorry, @youssef-lr We already have an internal engineer in this issue @cristipaval . Melvin seems to assign new one even if C+ comment is provided again. (I actually thought that was solved)

**Comment by abdulrahuman5196** [comment]
Waiting on @cristipaval 's review on this https://github.com/Expensify/App/issues/21322#issuecomment-1635893552

**Comment by abdulrahuman5196** [comment]
@jliexpensify @cristipaval is OOO for this week, should we check for another internal engineer for this?

**Comment by jliexpensify** [comment]
Haha I was just about to see if @youssef-lr might want to take it. Do you mind [helping us out](https://github.com/Expensify/App/issues/21322#issuecomment-1637763429) as Cristi is OOO?

**Comment by abdulrahuman5196** [comment]
Seems Melvin doesn't want to leave @youssef-lr out of this issue. 😂😂😂

**Comment by youssef-lr** [comment]
This [proposal](https://github.com/Expensify/App/issues/21322#issuecomment-1635893552) looks good to me as well. Assigning @Nodebrute.

**Comment by cristipaval** [comment]
I was ooo last week and today I've been catching up for what I missed. Thanks @youssef-lr for pushing this forward. I agree with the proposal 🙇

**Comment by abdulrahuman5196** [comment]
@Nodebrute I have made a comment regarding the same here https://expensify.slack.com/archives/C01GTK53T8Q/p1690262057767139?thread_ts=1690224959.854549&cid=C01GTK53T8Q

For now you can locally test by reverting the changes or making a rebase on older commit.

You could also try different browsers or from different IP?

**Comment by jliexpensify** [comment]
Not overdue!

**Comment by abdulrahuman5196** [comment]
>The PR that introduced the bug has been identified. Link to the PR:
The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not a regression. Followup of this PR https://github.com/Expensify/App/pull/20941#issuecomment-1602865492

>Determine if we should create a regression test for this bug.

YES

>If we decide to create a regression test for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


1) Go to a workspace
2) Request money from the workspace
3) Verify that you have the report preview where the actor is the workspace
4) Hover the workspace name on that report preview
5) Verify that you can see the tooltip

**Comment by jliexpensify** [comment]
Payment Summary:

- Bug Reporter: N/A (Cristi)
- Contributor: @Nodebrute $1000
- C+: @abdulrahuman5196 $1000

Is this correct? I don't think anyone is eligible for a bonus.

**Comment by jliexpensify** [comment]
@Nodebrute I have invited you (i think?) but can you add your full name from upworks to your GH profile? Thanks.

**Comment by abdulrahuman5196** [comment]
@jliexpensify  correct. Accepted the offer.

**Comment by jliexpensify** [comment]
Everyone paid, job closed!
